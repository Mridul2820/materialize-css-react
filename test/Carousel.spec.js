import React from 'react';
import { render } from '@testing-library/react';
import Carousel from '../src/Carousel';
import mocker from './helper/new-mocker';

describe('<Carousel />', () => {
  const images = [
    'https://lorempixel.com/250/250/nature/1',
    'https://lorempixel.com/250/250/nature/2',
    'https://lorempixel.com/250/250/nature/3',
    'https://lorempixel.com/250/250/nature/4'
  ];

  test('renders', () => {
    const { container } = render(<Carousel images={images} />);

    expect(container).toMatchSnapshot();
  });

  test('renders fixed items', () => {
    const fixedItem = <span>Do you rock!?</span>;
    const { container } = render(
      <Carousel images={images} fixedItem={fixedItem} />
    );

    expect(container).toMatchSnapshot();
  });

  test('renders centered images', () => {
    const { container } = render(<Carousel images={images} centerImages />);

    expect(container).toMatchSnapshot();
  });

  test('accepts className props', () => {
    const { container } = render(
      <Carousel images={images} className="center" />
    );

    expect(container).toMatchSnapshot();
  });

  test('accepts external `id` value', () => {
    const { container } = render(<Carousel images={images} carouselId="ID" />);

    expect(container).toMatchSnapshot();
  });

  describe('initialises', () => {
    const carouselInitMock = jest.fn();
    const carouselInstanceDestroyMock = jest.fn();
    const carouselMock = {
      init: (_, options) => {
        carouselInitMock(options);
        return {
          destroy: carouselInstanceDestroyMock
        };
      }
    };

    const restore = mocker('Carousel', carouselMock);

    beforeEach(() => {
      carouselInitMock.mockClear();
      carouselInstanceDestroyMock.mockClear();
    });

    afterAll(() => {
      restore();
    });

    test('uses default options if none are given', () => {
      render(<Carousel />);

      expect(carouselInitMock).toHaveBeenCalledWith({
        duration: 200,
        dist: -100,
        shift: 0,
        padding: 0,
        numVisible: 5,
        fullWidth: false,
        indicators: false,
        noWrap: false,
        onCycleTo: null
      });
    });

    test('handles full width sliders', () => {
      render(<Carousel images={images} options={{ fullWidth: true }} />);

      expect(carouselInitMock).toHaveBeenCalledWith({ fullWidth: true });
    });

    test('more options', () => {
      const options = { padding: 12, fullWidth: true, indicators: false };
      render(<Carousel images={images} options={options} />);
      expect(carouselInitMock).toHaveBeenCalledWith(options);
    });
  });
});
