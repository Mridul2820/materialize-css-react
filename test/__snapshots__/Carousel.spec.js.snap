// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Carousel /> accepts className props 1`] = `
<div>
  <div
    class="carousel center"
    id="Carousel-mockId"
  >
    <a
      class="carousel-item"
    >
      <img
        alt="https://lorempixel.com/250/250/nature/1"
        src="https://lorempixel.com/250/250/nature/1"
      />
    </a>
    <a
      class="carousel-item"
    >
      <img
        alt="https://lorempixel.com/250/250/nature/2"
        src="https://lorempixel.com/250/250/nature/2"
      />
    </a>
    <a
      class="carousel-item"
    >
      <img
        alt="https://lorempixel.com/250/250/nature/3"
        src="https://lorempixel.com/250/250/nature/3"
      />
    </a>
    <a
      class="carousel-item"
    >
      <img
        alt="https://lorempixel.com/250/250/nature/4"
        src="https://lorempixel.com/250/250/nature/4"
      />
    </a>
  </div>
</div>
`;

exports[`<Carousel /> accepts external \`id\` value 1`] = `
<div>
  <div
    class="carousel"
    id="ID"
  >
    <a
      class="carousel-item"
    >
      <img
        alt="https://lorempixel.com/250/250/nature/1"
        src="https://lorempixel.com/250/250/nature/1"
      />
    </a>
    <a
      class="carousel-item"
    >
      <img
        alt="https://lorempixel.com/250/250/nature/2"
        src="https://lorempixel.com/250/250/nature/2"
      />
    </a>
    <a
      class="carousel-item"
    >
      <img
        alt="https://lorempixel.com/250/250/nature/3"
        src="https://lorempixel.com/250/250/nature/3"
      />
    </a>
    <a
      class="carousel-item"
    >
      <img
        alt="https://lorempixel.com/250/250/nature/4"
        src="https://lorempixel.com/250/250/nature/4"
      />
    </a>
  </div>
</div>
`;

exports[`<Carousel /> renders 1`] = `
<div>
  <div
    class="carousel"
    id="Carousel-mockId"
  >
    <a
      class="carousel-item"
    >
      <img
        alt="https://lorempixel.com/250/250/nature/1"
        src="https://lorempixel.com/250/250/nature/1"
      />
    </a>
    <a
      class="carousel-item"
    >
      <img
        alt="https://lorempixel.com/250/250/nature/2"
        src="https://lorempixel.com/250/250/nature/2"
      />
    </a>
    <a
      class="carousel-item"
    >
      <img
        alt="https://lorempixel.com/250/250/nature/3"
        src="https://lorempixel.com/250/250/nature/3"
      />
    </a>
    <a
      class="carousel-item"
    >
      <img
        alt="https://lorempixel.com/250/250/nature/4"
        src="https://lorempixel.com/250/250/nature/4"
      />
    </a>
  </div>
</div>
`;

exports[`<Carousel /> renders centered images 1`] = `
<div>
  <div
    class="carousel"
    id="Carousel-mockId"
  >
    <a
      class="carousel-item valign-wrapper"
    >
      <img
        alt="https://lorempixel.com/250/250/nature/1"
        src="https://lorempixel.com/250/250/nature/1"
      />
    </a>
    <a
      class="carousel-item valign-wrapper"
    >
      <img
        alt="https://lorempixel.com/250/250/nature/2"
        src="https://lorempixel.com/250/250/nature/2"
      />
    </a>
    <a
      class="carousel-item valign-wrapper"
    >
      <img
        alt="https://lorempixel.com/250/250/nature/3"
        src="https://lorempixel.com/250/250/nature/3"
      />
    </a>
    <a
      class="carousel-item valign-wrapper"
    >
      <img
        alt="https://lorempixel.com/250/250/nature/4"
        src="https://lorempixel.com/250/250/nature/4"
      />
    </a>
  </div>
</div>
`;

exports[`<Carousel /> renders fixed items 1`] = `
<div>
  <div
    class="carousel"
    id="Carousel-mockId"
  >
    <div
      class="carousel-fixed-item center"
    >
      <span>
        Do you rock!?
      </span>
    </div>
    <a
      class="carousel-item"
    >
      <img
        alt="https://lorempixel.com/250/250/nature/1"
        src="https://lorempixel.com/250/250/nature/1"
      />
    </a>
    <a
      class="carousel-item"
    >
      <img
        alt="https://lorempixel.com/250/250/nature/2"
        src="https://lorempixel.com/250/250/nature/2"
      />
    </a>
    <a
      class="carousel-item"
    >
      <img
        alt="https://lorempixel.com/250/250/nature/3"
        src="https://lorempixel.com/250/250/nature/3"
      />
    </a>
    <a
      class="carousel-item"
    >
      <img
        alt="https://lorempixel.com/250/250/nature/4"
        src="https://lorempixel.com/250/250/nature/4"
      />
    </a>
  </div>
</div>
`;
